src/java/org/apache/cassandra/cql3/statements/AlterTableStatement.java

```diff
@@ -290,6 +290,10 @@ public void validate(ClientState state)
                                                                     columnFamily()));
                 break;
             case DROP_COMPACT_STORAGE:
+
+                if (!DatabaseDescriptor.enableDropCompactStorage())
+                    throw new InvalidRequestException("DROP COMPACT STORAGE is disabled. Enable in cassandra.yaml to use.");
```

Comment:
When we added the experimental flags for MVs and SASI indexes in CASSANDRA-14866 we also added a client warning for each of them, [here](https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/cql3/statements/schema/CreateViewStatement.java#L340) and [here](https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/cql3/statements/schema/CreateIndexStatement.java#L148). Shouldn't we warn users about `DROP COMPACT STORAGE` being experimatal too, especially given that it's allowed by default in the config?

Reply:
Of course the warning is not going to be that useful since in this case the damage is already done when the query is run. So probably it's better to just disable it by default.

Reply:
That was my reasoning once you have run the command and saw the warning it is effectively too late :-(